extension/utilities/lineItemIdToProductMapper.js

```diff
@@ -0,0 +1,15 @@
+const { RECHARGE_LINE_ITEM_ID_TO_PRODUCT_MAP } = require('../constants')
+
+const saveLineIdToProductIdMap = async (context, cartLineItems) => {
+  const mapping = cartLineItems
+    .filter(item => item.product && typeof item.product === 'object')
+    .map(item => ({ lineItemId: item.id, productId: item.product.id }))
+
+  return context.storage.device.set(RECHARGE_LINE_ITEM_ID_TO_PRODUCT_MAP, mapping)
```

Comment:
we should use the user storage here if the user is logged in

Reply:
done